util/threadpool_imp.h

```diff
@@ -33,7 +33,9 @@ class ThreadPoolImpl : public ThreadPool {
   void JoinAllThreads() override;
 
   // Set the number of background threads that will be executing the
-  // scheduled jobs.
+  // scheduled jobs. (This function will reset reserved_threads to 0,
+  // reset is only triggered when total_threads_limit changes, typically,
+  // (num > total_threads_limit) || (num < total_threads_limit && allow_reduce)
```

Comment:
Comments not needed?

Reply:
Ohh, thanks a lot for catching this! This was an old comment for my earlier implementation. I will remove this.